src/main/java/io/github/cottonmc/cotton/mixin/TorchBlockMixin.java

```diff
@@ -0,0 +1,39 @@
+package io.github.cottonmc.cotton.mixin;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.TorchBlock;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BoundingBox;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.world.ViewableWorld;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(TorchBlock.class)
```

Comment:
Maybe add a comment here explaining for which tweak we need to modify the torch. Something like "Allows the placement of torches based on the collision of the block below."

Reply:
I added some comments.